core/global/plugin_loader.gd

```diff
@@ -375,71 +376,108 @@ func initialize_plugin(plugin_id) -> int:
 	return OK
 
 
-# Looks in the user plugins directory for plugin json files and loads them.
+# Looks in both user plugins directory (user://plugins) and resource plugins directory (res://plugins) 
+# for plugin zip files or directories with plugin.json and loads them.
 func _load_plugins() -> void:
-	var dir = DirAccess.open(PLUGINS_DIR)
-	if not dir:
-		logger.debug("Unable to open plugin directory!")
-		return
-
-	# Iterate through all plugins
-	dir.list_dir_begin()
-	var file_name = dir.get_next()
-	while file_name != "":
-		# Skip any non-plugins
-		if not file_name.ends_with(".zip"):
-			file_name = dir.get_next()
-			continue
-
-		# Read the plugin metadata
-		logger.debug("Found plugin: " + file_name)
-		var meta = _load_plugin_meta("/".join([PLUGINS_DIR, file_name]))
-		if meta == null:
-			logger.warn("%s failed to load" % file_name)
-			file_name = dir.get_next()
-			continue
-
-		# Validate the plugin metadata
-		if not _is_valid_plugin_meta(meta):
-			logger.warn("%s has invalid plugin JSON" % file_name)
-			file_name = dir.get_next()
-			continue
-
-		# Ensure the plugin is compatible
-		if not _is_compatible_version(meta["plugin.min-api-version"], PLUGIN_API_VERSION):
-			logger.warn("%s is not compatible with this plugin API verion" % file_name)
-			file_name = dir.get_next()
+	var plugin_dirs := [
+		PLUGINS_DIR,         # user://plugins
+		LOADED_PLUGINS_DIR   # res://plugins
+	]
+	
+	for plugin_dir_path in plugin_dirs:
```

Comment:
this change adds additional nesting. To make the code easier to read and maintain we should retain the old logic in a new function that takes the directory it is looking for as an argument.

i.e.
```
func _load_plugins() -> void:
	var plugin_dirs := [
		PLUGINS_DIR,         # user://plugins
		LOADED_PLUGINS_DIR   # res://plugins
	]

	for plugin_dir_path in plugin_dirs:
	        _load_plugins_from_dir(plugin_dir_path)

func _load_plugins_from_dir(plugin_dir_path: String) -> void:
        var dir = DirAccess.open(plugin_dir_path
        ...
```